packages/kit/test/apps/basics/test/server.test.js

```diff
@@ -192,13 +192,31 @@ test.describe('Endpoints', () => {
 	test('OPTIONS handler', async ({ request }) => {
 		const url = '/endpoint-output/options';
 
-		var response = await request.fetch(url, {
+		const response = await request.fetch(url, {
 			method: 'OPTIONS'
 		});
 
 		expect(response.status()).toBe(200);
 		expect(await response.text()).toBe('ok');
 	});
+
+	test('catch-all handler', async ({ request }) => {
+		const url = '/endpoint-output/all';
+
+		let response = await request.fetch(url, {
+			method: 'GET'
+		});
+
+		expect(response.status()).toBe(200);
+		expect(await response.text()).toBe('ok');
+
+		response = await request.fetch(url, {
+			method: 'OPTIONS'
+		});
+
+		expect(response.status()).toBe(200);
+		expect(await response.text()).toBe('catch-all');
```

Comment:
On the linked issue I only found one comment about custom methods, so can't tell if people want it to to be like "if I don't have a PATCH, PUT etc. use the all handler"  or more that they want to use custom HTTP methods.

If the latter there should be another test case with a non-standard method and this https://github.com/sveltejs/kit/blob/2a7dacbe4e3ae7a810f0e2a516df76357c1a78c6/packages/kit/src/runtime/server/endpoint.js#L72-L86 probably needs to be updated as well.

Reply:
> or more that they want to use custom HTTP methods.

I had the same question about custom HTTP methods and posted in the maintainer's chat.
Here's one of the replies I got https://discord.com/channels/457912077277855764/571775594002513921/1099684470392639508
TLDR: at the moment we're only allowing a subset of most-used HTTP methods.
https://github.com/sveltejs/kit/pull/9752/files will explicitly return a 405 for disallowed methods and fix another issue as well.